corehq/apps/registration/forms.py

```diff
@@ -131,20 +131,27 @@ class DomainRegistrationForm(forms.Form):
     """
     Form for creating a domain for the first time
     """
+    max_name_length = 25
+
     org = forms.CharField(widget=forms.HiddenInput(), required=False)
-    domain_name = forms.CharField(label=_('Project Name:'), max_length=25,
-                                  help_text=_("Project name cannot contain spaces."))
+    hr_name = forms.CharField(label=_('Project Name:'), max_length=max_name_length)
+    domain_name = forms.CharField(widget=forms.HiddenInput(), required=False)
     domain_type = forms.CharField(widget=forms.HiddenInput(), required=False,
                                   initial='commcare')
 
     def clean_domain_name(self):
-        data = self.cleaned_data['domain_name'].strip().lower()
-        if not re.match("^%s$" % new_domain_re, data):
-            raise forms.ValidationError('Only lowercase letters and numbers allowed. Single hyphens may be used to separate words.')
-
+        data = self.cleaned_data['hr_name'].strip().lower()
+        data = re.sub(r'[^0-9a-z]+', '-', data)
         conflict = Domain.get_by_name(data) or Domain.get_by_name(data.replace('-', '.'))
         if conflict:
-            raise forms.ValidationError('Project name already taken---please try another')
+            counter = 0
+            test_name = data
+            while conflict:
+                counter = counter + 1
+                suffix = "-" + str(counter)
+                test_name = data[:self.max_name_length - len(suffix)] + suffix
+                conflict = Domain.get_by_name(test_name) or Domain.get_by_name(test_name.replace('-', '.'))
```

Comment:
Changed to use ES. It's not down to one query for names with a lot of conflicts, but I figured that by the time we get to a point where there are that many conflicts, pulling in all domain names is going to be prohibitively expensive.